infra/scripts/replace-running-components.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+# Copyright 2018 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+BASE_DIR=$(dirname $(readlink -f "$BASH_SOURCE"))
+VIC_DIR=$(dirname $(readlink -f $BASE_DIR/..))
+
+# Grab vSphere's thumbprint for calling vic-machine
+function get-thumbprint () {
+    govc about.cert | grep "SHA-1 Thumbprint" | awk '{print $NF}'
+}
+
+# Run the command given on the VCH instead of locally
+function on-vch() {
+    sshpass -ppassword ssh -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no root@$VCH_IP -C $@ 2>/dev/null
+}
+
+# SCPs the component in $1 to the VCH, plops it in place, and brutally kills the previous running process
+function replace-component() {
+    sshpass -p 'password' scp -oUserKnownHostsFile=/dev/null -oStrictHostKeyChecking=no $VIC_DIR/bin/$1 root@$VCH_IP:/tmp/$1
+    on-vch mv /sbin/$1 /tmp/old-$1
+    on-vch mv /tmp/$1 /sbin/$1
+    on-vch chmod 755 /sbin/$1
+    pid=$(on-vch ps -e --format='pid,args' | grep $1 | grep -v grep | awk '{print $1}')
+    on-vch kill -9 $pid
+    on-vch rm -f /tmp/old-$1
+}
+
+function vch-name-is-ambiguous () {
+    [ $($VIC_DIR/bin/vic-machine-linux ls --target=$target --user=$username --password=$password --thumbprint=$(get-thumbprint)  | grep $VIC_NAME | wc -l) -ne 1 ] && return 0 || return 1
+}
+
+# Check GOVC vars
+if [[ ! $(govc ls 2>/dev/null) ]]; then
+    echo "ERROR:"
+    echo "GOVC environment variables are required to use this command. Set the necessary variables to allow govc to connect to your vSphere deployment:";
+    echo "GOVC_USERNAME: username on vSphere target"
+    echo "GOVC_PASSWORD: password on vSphere target"
+    echo "GOVC_URL: IP or FQDN of your vSphere target"
+    echo "GOVC_INSECURE: set to 1 to disable tls verify when using govc to talk to vSphere"
+    exit 1
+fi
+
+# Check for our one required argument
+if [[ ! -v VIC_NAME ]] && [[ ! -v VIC_ID ]]; then
+    echo "Please set one of the following environment variables to specify the VCH which you would like to reconfigure:"
+    echo "VIC_NAME: name of VCH; matches --name argument for vic-machine"
+    echo "VIC_ID: ID of VCH, as displayed in output of vic-machine ls"
+    exit 1
+fi
+
+username=$(govc env | grep GOVC_USERNAME | cut -d= -f2)
```

Comment:
This will be problem for those of us who define GOVC_URL="user:pass@targeturl".  We could try to grab user/pass from GOVC_URL if GOVC_USERNAME and GOVC_PASSWORD are not defined, but I see you do echo usage info explaining this.

Reply:
`govc env` should break those out already if you only assign `GOVC_URL` so this should be safe from this concern. By using `govc env` it should always have all of the required fields for GOVC populated for consumption, which should avoid a "something isn't set right" style of error.

Reply:
oh, interesting.  I didn't know about this option.